File: database/src/main/java/bao/webapp/service/UserServiceImpl.java
package bao.webapp.service;

import java.util.ArrayList;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import bao.webapp.model.User;
import bao.webapp.repository.UserRepository;

@Service
public class UserServiceImpl implements UserService {

	@Autowired
	private UserRepository userRepository;

	public List<User> list() {
		List<User> all = new ArrayList<>();

		for (User user : this.userRepository.findAll())
			all.add(user);

		return all;
	}

	@Override
	public User findById(long id) {
		return this.userRepository.findById(id).get();
	}

	@Override
	public User findByUsername(String username) {
		return this.userRepository.findByUserName(username);
	}

	@Override
	public void create(User user) {
		this.userRepository.save(user);
	}
	
}

File: database/src/main/kotlin/com/ossp/database/service/UserServiceImpl.kt
package com.ossp.database.service

import com.ossp.database.model.User
import com.ossp.database.repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserServiceImpl : UserService {

    @Autowired
    private val userRepository: UserRepository? = null

    override fun list(): List<User> {
        val all = ArrayList<User>()

        userRepository?.let { repository ->
            for (user in repository.findAll()) {
                all.add(user)
            }
        }

        return all
    }

    override fun findById(id: Long?): User? {
        id?.let { userId ->
            return this.userRepository?.findById(userId)?.get()
        } ?: run {
            return null
        }
    }

    override fun findByUsername(username: String): User? {
        return this.userRepository?.findByUserName(username)
    }

    override fun create(user: User): User? {
        this.userRepository?.save(user)
        return userRepository?.findByUserName(user.userName)
    }
}